README.md

```diff
@@ -1,90 +1,186 @@
+- Validate with `terraform validate` (enforced by CI)
+- Format with `terraform format` (enforced by CI)
+- Run `terraform plan` locally to inspect the potential changes (also run in CI)
+- Thoroughly review pull requests
+- (TODO) Lint with [TFLint](https://github.com/terraform-linters/tflint)
+- (TODO) Security Scan with [TFSEC](https://github.com/tfsec/tfsec)
 
-### `sns`
-Contains the configuration to create an AWS SNS instance with associated users/policies.
+# CI/CD
+Changes in the Terraform configuration are applied using [Github Actions](https://docs.github.com/en/free-pro-team@latest/actions) according to the [`terraform` workflow](https://github.com/18F/federalist-infra/blob/main/.github/workflows/terraform.yml). When a Github pull request is created agains the default branch (`main`), the `terraform` job is run for *each* configured environment (`staging`, `production`) with the results of the corresponding `terraform plan` added as a comment to the pull request. This output should be reviewed in detail before the pull request is approved, `terraform apply` is only run when the pull request is merged to the default branch.
+
+Note: Only the `staging` and `production` *root* modules should be run in CI.
+
+### Environment Variables
+#### Terraform Backend
+These credentials correspond to the `terraform-backend` user configured in the AWS Admin GovCloud account.
+- `BACKEND_AWS_ACCESS_KEY_ID`
+- `BACKEND_AWS_SECRET_ACCESS_KEY`
+- `BACKEND_AWS_DEFAULT_REGION`
+
+#### AWS
+These credentials correspond to the `terraform-user` user configured in the specified platform AWS Admin account.
+- `TF_VAR_aws_access_key_govcloud`
+- `TF_VAR_aws_secret_key_govcloud`
+- `TF_VAR_aws_access_key_commercial`
+- `TF_VAR_aws_secret_key_commercial`
+
+#### Cloud Foundry
+These correspond to the space deployer credentials in each environment: (`federalist-staging-deployer-circle`, `federalist-production-deployer-circle`)
+- `CF_USER_STAGING`
+- `CF_PASSWORD_STAGING`
+- `CF_USER_PRODUCTION`
+- `CF_PASSWORD_PRODUCTION`
+
+#### CircleCI
+CircleCI API Key
+- `TF_VAR_circleci_api_key`
+
+# Boostrapping
+To ensure that we can use least-privilege credentials when provisioning resources with Terraform, it is necessary to have some bootstrapping steps that are run once with privileged credentials in a local environment.
+
+### Bootstrapping the backend
+This only needs to be done once, and only when starting completely from scratch. Since the Terraform configuration includes the resources user for storing the Terraform state, we must create the resources first using local state, then add the backend configuration which makes use of those resources. Terraform is smart enough to recognize when the backend configuration chances
```

Comment:
```suggestion
This only needs to be done once, and only when starting completely from scratch. Since the Terraform configuration includes the resources user for storing the Terraform state, we must create the resources first using local state, then add the backend configuration which makes use of those resources. Terraform is smart enough to recognize when the backend configuration changes.
```